onnxruntime/core/providers/coreml/builders/impl/reshape_op_builder.cc

```diff
@@ -94,6 +96,18 @@ bool ReshapeOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializer
     return false;
   }
 
+  // CoreML reshape does not support 0 as dimension
+  NodeAttrHelper helper(node);
+  bool allow_zero = helper.Get("allowzero ", 0) == 1;
```

Comment:
nit: could be const
similar comment for some bools in the op_support_checker.cc updates